src/DocumentPicker.Plugin/Shared/Picker/FilePickerResult.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+
+namespace DocumentConverter.Plugin.Shared.Picker
+{
+    public class FilePickerResult : IDisposable
+    {
+        public string FileName { get; set; }
+        public string FullPath { get; set; }
+
+        public FilePickerResult(string fileName, string fullPath)
+        {
+            FileName = fileName;
+            FullPath = fullPath;
+        }
+        
+        public Action OnDispose { get; set; } = (() => { return; });
```

Comment:
I would suggest you ensure this action can never be assigned from any user of your API.
So make it private and provide the action in the ctor.

However, I would think it is nicer to just delete the file you are sharing?
Is there any reason you are deleting _all_ of the files?
Can't there be a case where you delete more files than you actually want? (Convince me if I am wrong ;-))

Reply:
the picked file is saved in a created directory "DocumentConverter", i just wanted to always delete the whole folder just to have a cleaner file system.

Reply:
That's ok then.
However, anyone could still override the "OnDispose" property - this is a design flaw.
Better make the constructor internal, so only your library can create results and pass the Action in the constructor